uncrustify.cfg

```diff
@@ -0,0 +1,83 @@
+# indent and align with tabs, only uses spaces if tab is too large
+indent_with_tabs = 2
+indent_cmt_with_tabs = true
+
+# No newlines between type name and brace
+nl_enum_brace = remove
+nl_struct_brace = remove
+nl_union_brace = remove
+
+# No newlines before block
+nl_assign_brace = remove
+nl_fcall_brace = remove
+nl_if_brace = remove
+nl_brace_else = remove
+nl_elseif_brace = remove
+nl_for_brace = remove
+nl_while_brace = remove
+nl_do_brace = remove
+nl_brace_while = remove
+
+# Consistent newline behavior in function definitions
+nl_fdef_brace = add
+nl_func_type_name = remove
+
+# Use consistent spacing
+sp_arith_additive = force
+sp_assign = force
+sp_enum_paren = force
+sp_enum_assign = force
+sp_bool = force
+sp_compare = force
+sp_inside_paren = remove
+sp_paren_paren = remove
+sp_cparen_oparen = remove
+sp_paren_brace = force
+sp_brace_brace = remove
+sp_before_ptr_star = force
+sp_between_ptr_star = remove
+sp_after_ptr_star = remove
+sp_after_byref = remove
+sp_before_sparen = force
+sp_inside_sparen = remove
+sp_inside_for = remove
+sp_after_sparen = force
+sp_sparen_brace = force
+sp_do_brace_open = force
+sp_while_paren_open = force
+sp_before_semi_for = remove
+sp_after_semi_for = force
+sp_inside_square = remove
+sp_after_comma = force
+sp_after_cast = ignore  # XXX: We're not very consistent with this, do we want to be?
```

Comment:
I would like the setting "add" here (or "force", what's the difference?). But it's not how the current code is formatted, mostly. I suggest leaving it as is, with the comment, and we can address it as a follow-up.

Reply:
Sounds good. 

> "add" here (or "force", what's the difference?)

FYI: `force` means make it 1 space. `add` means add 1 space when there are none, but if there is already spacing leave as is So if you have 3 spaces, then force makes it 1, but add leaves as 3.

Reply:
Ok, let's look at that later.